From 2f0417388e4b64381f5f6a0e63d361e5ce80387f Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Mon, 5 Jun 2023 08:38:46 +0200
Subject: [PATCH]  treble : add dynamic securize , and some extra  functions

- Add Dynamic securize
- Add Autorun
- Add NoLog
- Add Restart sysui
- Add Dump Logs
- Add no stock apps (vendor)
- Add Double Tape 2W (qin)
- Remove gnss_watchlssd_thirdparty (agps)
---
 base.mk             |  1 +
 phh-on-data.sh      | 15 +++++---
 phh-prop-handler.sh | 87 ++++++++++++++++++++++++++++++++++++++++++++-
 rw-system.sh        | 22 ++++++------
 secure.sh           | 16 +++++++++
 vndk.rc             | 31 ++++++++++++++--
 6 files changed, 153 insertions(+), 19 deletions(-)
 create mode 100644 secure.sh

diff --git a/base.mk b/base.mk
index 59d7e84..92715b5 100644
--- a/base.mk
+++ b/base.mk
@@ -87,6 +87,7 @@ PRODUCT_PACKAGES += \
 	android.hardware.nfc@1.2-service \
 
 PRODUCT_COPY_FILES += \
+	device/phh/treble/secure.sh:system/phh/secure.sh \
 	device/phh/treble/rw-system.sh:system/bin/rw-system.sh \
 	device/phh/treble/phh-on-data.sh:system/bin/phh-on-data.sh \
 	device/phh/treble/phh-prop-handler.sh:system/bin/phh-prop-handler.sh \
diff --git a/phh-on-data.sh b/phh-on-data.sh
index 472d16e..34ad54b 100644
--- a/phh-on-data.sh
+++ b/phh-on-data.sh
@@ -3,25 +3,32 @@
 vndk="$(getprop persist.sys.vndk)"
 [ -z "$vndk" ] && vndk="$(getprop ro.vndk.version |grep -oE '^[0-9]+')"
 
-if getprop persist.sys.phh.no_vendor_overlay |grep -q true;then
+if getprop persist.sys.phh.no_vendor_overlay |grep -q true; then
 	for part in odm vendor;do
 		mount /mnt/phh/empty_dir/ /$part/overlay
 	done
 fi
 
-if getprop persist.sys.phh.caf.media_profile |grep -q true;then
+if getprop persist.sys.phh.no_stock_apps |grep -q true; then
+	for part in odm vendor; do
+		mount /mnt/phh/empty_dir/ /$part/overlay
+		mount /mnt/phh/empty_dir/ /$part/app
+	done
+fi
+
+if getprop persist.sys.phh.caf.media_profile |grep -q true; then
     setprop media.settings.xml "/vendor/etc/media_profiles_vendor.xml"
 fi
 
 
 minijailSrc=/system/system_ext/apex/com.android.vndk.v28/lib/libminijail.so
 minijailSrc64=/system/system_ext/apex/com.android.vndk.v28/lib64/libminijail.so
-if [ "$vndk" = 27 ];then
+if [ "$vndk" = 27 ]; then
     mount $minijailSrc64 /vendor/lib64/libminijail_vendor.so
     mount $minijailSrc /vendor/lib/libminijail_vendor.so
 fi
 
-if [ "$vndk" = 28 ];then
+if [ "$vndk" = 28 ]; then
     mount $minijailSrc64 /vendor/lib64/libminijail_vendor.so
     mount $minijailSrc /vendor/lib/libminijail_vendor.so
     mount $minijailSrc64 /system/lib64/vndk-28/libminijail.so
diff --git a/phh-prop-handler.sh b/phh-prop-handler.sh
index 81360cc..0b4cc9d 100644
--- a/phh-prop-handler.sh
+++ b/phh-prop-handler.sh
@@ -70,6 +70,16 @@ restartAudio() {
     setprop ctl.restart audio-hal-2-0
 }
 
+
+if [ "$1" == "persist.sys.phh.qin.dt2w" ];then
+    if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
+        exit 1
+    fi
+
+    echo "$prop_value" > /sys/devices/platform/soc/soc:ap-apb/70800000.i2c/i2c-3/3-0038/fts_gesture_mode
+    exit
+fi
+
 if [ "$1" == "persist.sys.phh.xiaomi.dt2w" ]; then
     if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
         exit 1
@@ -211,6 +221,79 @@ if [ "$1" == "persist.sys.phh.disable_soundvolume_effect" ];then
     exit
 fi
 
+
+#securize
+if [ "$1" == "persist.sys.phh.securize" ]; then
+    if [[ "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/secure ]; then
+        mkdir -p /data/adb/phh
+        cp /system/phh/secure.sh /data/adb/phh/secure
+    fi
+    /system/bin/sh /data/adb/phh/secure
+    exit
+fi
+
+#autorun
+if [ "$1" == "persist.sys.phh.autorun" ]; then
+    if [[ "$prop_value" != "true" && "$prop_value" != "false" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/run ]; then
+        mkdir -p /data/adb/phh
+        touch /data/adb/phh/run
+    fi
+    /system/bin/sh /data/adb/phh/run
+    exit
+fi
+
+#nolog
+if [ "$1" == "persist.sys.phh.nolog" ]; then
+    if [[ "$prop_value" != "false" && "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        setprop ctl.stop logd
+        setprop ctl.stop traced
+        setprop ctl.stop traced_probes
+    else
+        setprop ctl.start traced_probes
+        setprop ctl.start traced
+        setprop ctl.start logd
+    fi
+    exit
+fi
+
+#restart_sysui
+if [ "$1" == "sys.phh.restart_sysui" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        pkill systemui
+    fi
+    exit
+fi
+
+#dump_logs
+if [ "$1" == "sys.phh.dump_logs" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        logcat -b all -d > /sdcard/logs.txt
+    fi
+    exit
+fi
+
+
+
 #monosim
 if [ "$1" == "persist.sys.phh.monosim" ]; then
     if [[ "$prop_value" != "false" && "$prop_value" != "true" ]]; then
@@ -241,6 +324,8 @@ if [ "$1" == "persist.sys.phh.monosim" ]; then
     exit
 fi
 
+
+# disable audio_hal
 if [ "$1" == "persist.bluetooth.system_audio_hal.enabled" ]; then
     if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
         exit 1
@@ -257,4 +342,4 @@ if [ "$1" == "persist.bluetooth.system_audio_hal.enabled" ]; then
     fi
     restartAudio
     exit
-fi
\ No newline at end of file
+fi
diff --git a/rw-system.sh b/rw-system.sh
index 5dc493f..782a157 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -575,10 +575,6 @@ if getprop ro.hardware | grep -qF exynos; then
     setprop debug.sf.latch_unsignaled 1
 fi
 
-if getprop ro.product.model | grep -qF ANE; then
-    setprop debug.sf.latch_unsignaled 1
-fi
-
 if getprop ro.vendor.product.device | grep -q -e nora -e rhannah; then
     setprop debug.sf.latch_unsignaled 1
 fi
@@ -602,16 +598,18 @@ if getprop ro.vendor.build.fingerprint | grep -iq -e Redmi/rosemary \
     mount -o bind /system/phh/rosemary-excluded-input-devices.xml /system/etc/excluded-input-devices.xml
 fi
 
+# Iceows patch - NFC NXP conf remove, you must set by specify model patch
+# remove gnss_watchlssd_thirdparty to avoid log spam
 if getprop ro.vendor.build.fingerprint | grep -iq -E -e 'huawei|honor' || getprop persist.sys.overlay.huawei | grep -iq -E -e 'true'; then
-    p=/product/etc/nfc/libnfc_nxp_*_*.conf
-    mount -o bind "$p" /system/etc/libnfc-nxp.conf ||
-        mount -o bind /product/etc/libnfc-nxp.conf /system/etc/libnfc-nxp.conf || true
 
-    p=/product/etc/nfc/libnfc_brcm_*_*.conf
-    mount -o bind "$p" /system/etc/libnfc-brcm.conf ||
-        mount -o bind /product/etc/libnfc-nxp.conf /system/etc/libnfc-nxp.conf || true
+   setprop debug.sf.latch_unsignaled 1
+
+    # Set proper /sdcard permissions to avoid dead storage
+   chown media_rw:media_rw /data/media/0
+   chmod 0770 /data/media/0
 
-    mount -o bind /system/phh/libnfc-nci-huawei.conf /system/etc/libnfc-nci.conf
+    # Disable this watchlss extension to fix logspams and dead gps
+    mount -o bind /system/phh/empty /vendor/bin/gnss_watchlssd_thirdparty
 fi
 
 if getprop ro.vendor.build.fingerprint | grep -qE -e ".*(crown|star)[q2]*lte.*" -e ".*(SC-0[23]K|SCV3[89]).*" && [ "$vndk" -lt 28 ]; then
@@ -739,7 +737,7 @@ copyprop() {
         resetprop_phh "$1" "$(getprop "$2")"
     fi
 }
-if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ];then
+if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ] || [ -f /data/adb/phh/secure ];then
     copyprop ro.build.device ro.vendor.build.device
     copyprop ro.system.build.fingerprint ro.vendor.build.fingerprint
     copyprop ro.bootimage.build.fingerprint ro.vendor.build.fingerprint
diff --git a/secure.sh b/secure.sh
new file mode 100644
index 0000000..98027e7
--- /dev/null
+++ b/secure.sh
@@ -0,0 +1,16 @@
+(getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
+    [ -n "$v" ] && resetprop_phh ro.build.version.security_patch "$v"
+done
+resetprop_phh ro.build.tags release-keys
+resetprop_phh ro.boot.vbmeta.device_state locked
+resetprop_phh ro.boot.verifiedbootstate green
+resetprop_phh ro.boot.flash.locked 1
+resetprop_phh ro.boot.veritymode enforcing
+resetprop_phh ro.boot.warranty_bit 0
+resetprop_phh ro.warranty_bit 0
+resetprop_phh ro.debuggable 0
+resetprop_phh ro.secure 1
+resetprop_phh ro.build.type user
+resetprop_phh ro.build.selinux 0
+resetprop_phh ro.adb.secure 1
+setprop ctl.restart adbd
diff --git a/vndk.rc b/vndk.rc
index b5fc7fb..e4df69c 100644
--- a/vndk.rc
+++ b/vndk.rc
@@ -52,11 +52,38 @@ on property:persist.sys.phh.disable_soundvolume_effect=*
 on property:persist.sys.phh.monosim=*
     exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.monosim"
 
+on property:persist.bluetooth.system_audio_hal.enabled=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.bluetooth.system_audio_hal.enabled"
+
 on property:ro.sf.lcd_density=*
     restart surfaceflinger
 
-on property:persist.bluetooth.system_audio_hal.enabled=*
-    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.bluetooth.system_audio_hal.enabled"
+on property:persist.sys.phh.qin.dt2w=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.qin.dt2w"
+
+on property:persist.sys.phh.disable_a2dp_offload=0
+    setprop persist.sys.phh.disable_a2dp_offload false
+
+on property:persist.sys.phh.disable_a2dp_offload=1
+    setprop persist.sys.phh.disable_a2dp_offload true
+
+on property:persist.sys.phh.disable_a2dp_offload=*
+    setprop persist.bluetooth.bluetooth_audio_hal.disabled ${persist.sys.phh.disable_a2dp_offload}
+
+on property:persist.sys.phh.securize=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.securize"
+
+on property:persist.sys.phh.autorun=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.autorun"
+
+on property:persist.sys.phh.nolog=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.nolog"
+
+on property:sys.phh.restart_sysui=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.restart_sysui"
+
+on property:sys.phh.dump_logs=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.dump_logs"
 
 on property:init.svc.ril-proxy=stopped && property:persist.sys.phh.restart_ril=true
     start ril-proxy
-- 
2.25.1

